src/common/string_utils.cpp

```diff
@@ -31,6 +31,12 @@ std::vector<std::string> StringUtils::splitBySpace(const std::string& input) {
     return result;
 }
 
+std::string StringUtils::extractSubstring(std::string& str, char delimiter) {
```

Comment:
I don't think the function name is accurate here. As a general substring extractor (in `StringUtils`), it seems to assume that there are exactly two delimiters, and extract the substring between them, which only covers a subset of general substring extracting.

For now, this is only for the special usage that you are extracting path from the COPY statement. 
I would just put it under `CSVToParquetConverter`, and call it `extractPath(std::string& str)` maybe.